include/seqan3/alignment/pairwise/detail/policy_affine_gap_recursion_simd.hpp

```diff
@@ -0,0 +1,63 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::policy_affine_gap_recursion_simd.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <tuple>
+
+#include <seqan3/alignment/pairwise/detail/policy_affine_gap_recursion.hpp>
+#include <seqan3/core/simd/simd_algorithm.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Implements the alignment recursion function for the vectorised alignment algorithm using affine gap costs.
+ * \ingroup pairwise_alignment
+ *
+ * \copydetails seqan3::detail::policy_affine_gap_recursion
+ */
+template <typename alignment_configuration_t>
+class policy_affine_gap_recursion_simd : protected policy_affine_gap_recursion<alignment_configuration_t>
+{
+protected:
+    //!\brief The type of the base class.
+    using base_policy_t = policy_affine_gap_recursion<alignment_configuration_t>;
+    // Import the score type from the base.
+    using typename base_policy_t::score_type;
+
+    // Import parameters.
+    using base_policy_t::gap_extension_score;
+    using base_policy_t::gap_open_score;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    policy_affine_gap_recursion_simd() = default; //!< Defaulted.
+    policy_affine_gap_recursion_simd(policy_affine_gap_recursion_simd const &) = default; //!< Defaulted.
+    policy_affine_gap_recursion_simd(policy_affine_gap_recursion_simd &&) = default; //!< Defaulted.
+    policy_affine_gap_recursion_simd & operator=(policy_affine_gap_recursion_simd const &) = default; //!< Defaulted.
+    policy_affine_gap_recursion_simd & operator=(policy_affine_gap_recursion_simd &&) = default; //!< Defaulted.
+    ~policy_affine_gap_recursion_simd() = default; //!< Defaulted.
+
+    //!\copydoc seqan3::detail::policy_affine_gap_recursion::policy_affine_gap_recursion
+    explicit policy_affine_gap_recursion_simd(alignment_configuration_t const & config)
+    {
+        // Get the gap scheme from the config or choose -1 and -10 as default.
+        auto && selected_gap_scheme = config.template value_or<align_cfg::gap>(gap_scheme{gap_score{-1},
```

Comment:
> Why did you use here an extra class instead of making policy_affine_gap_recursion itself simd aware?

Because it is the better design. Later if we need the traceback I need to add thousands of ugly if constexpr things or a specific methods to check if I am a simd type or not. Also I need to implement the kernels a little different. In this case I only overload what is necessary by this policy.

Reply:
I don't know about your argument, you would have ONE additional constexpr right now.

This file is 63 lines for 2 more lines in the base class which is already simd-aware except the constructor.
I would approach that the other way around, if it turns out that the policy_affine_gap_recursion class must be split, let's do it when you need it.

Reply:
I really don't see the point. I have a very different point on the design on this and I don't want to change this. But oh my. Let's do it to spare 61 lines for 14 days.